mlrun/datastore/sources.py

```diff
@@ -118,7 +118,10 @@ def to_spark_df(self, session, named_view=False, time_field=None, columns=None):
             if named_view:
                 df.createOrReplaceTempView(self.name)
             return self._filter_spark_df(df, time_field, columns)
-        raise NotImplementedError()
+        raise NotImplementedError(
+            f"Conversion of a source of type '{type(self).__name__}' "
+            "to a Spark dataframe is not possible, as this operation is not supported"
```

Comment:
```suggestion
            "to a Spark dataframe is not possible, as this operation is not supported by Spark"
```

Reply:
Pretty sure spark supports reading from SQL, but `SQLSource` doesn't support  spark. So it's probably not right to blame spark, at least not in all cases.

Reply:
Following Gal's comment, I suggest we limit my change to KafkaSource, and leave your comment for the rest of the sources.